src/main/java/org/wildfly/security/auth/client/ElytronXmlParser.java

```diff
@@ -126,16 +129,31 @@
 
     private static final Supplier<Provider[]> DEFAULT_PROVIDER_SUPPLIER = ProviderUtil.aggregate(ELYTRON_PROVIDER_SUPPLIER, Security::getProviders);
 
-    static final String NS_ELYTRON_1_0 = "urn:elytron:1.0";
-    static final String NS_ELYTRON_CLIENT_1_1 = "urn:elytron:client:1.1";
+    static final Map<String, Version> KNOWN_NAMESPACES;
 
-    private static final Set<String> KNOWN_NAMESPACES;
+    private static final class Version {
+        final int major;
+        final int minor;
+
+        Version(int major, int minor) {
```

Comment:
Maybe we would be better with a single enum that encapsulates the namespace as well as the version information, we can still add it to a map with the namespace as a key but at least the population of the map would not be a manual correlation.